### PR TITLE
Add finalised ingress config

### PR DESCRIPTION
**Pull request by kierenevans**
Uses our current best practise configuration for using the NGINX Ingress Controller.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM quay.io/continuouspipe/symfony-php7.1-nginx:stable
+FROM quay.io/continuouspipe/symfony-php7.1-nginx:latest
```
Comment (sroze) [inline Dockerfile:1]: Why does it need that?

File: continuous-pipe.yml [hunk 2]
```diff
@@ -7,34 +7,57 @@ variables:
...
         web:
+          endpoints:
+            - name: web
+              cloud_flare_zone:
+                zone_identifier: ${CLOUD_FLARE_ZONE}
+                authentication:
+                    email: ${CLOUD_FLARE_EMAIL}
+                    api_key: ${CLOUD_FLARE_API_KEY}
+                proxied: true
+                record_suffix: '-sfdemo.continuouspipe.net'
```
Comment (sroze) [inline continuous-pipe.yml:44]: You don't need that while using it with the `ingress` :)

File: continuous-pipe.yml [hunk 3]
```diff
@@ -7,34 +7,57 @@ variables:
...
+                value: "true"
+              - name: AUTH_HTTP_HTPASSWD
+                value: ${AUTH_HTTP_HTPASSWD}
+              - name: AUTH_IP_WHITELIST_ENABLED
+                value: "true"
+              - name: AUTH_IP_WHITELIST
+                value: ${AUTH_IP_WHITELIST}
+              - name: TRUSTED_REVERSE_PROXIES
+                value: ${TRUSTED_REVERSE_PROXIES}
+              - name: WEB_REVERSE_PROXIED
```
Comment (sroze) [inline continuous-pipe.yml:62]: Why do we need that?

File: Dockerfile (hunk 1 above)
Comment (kierenevans) [reply Dockerfile:1]: stable doesn't have trusted ip support. Need to test a release for new stable. I think it works.

File: continuous-pipe.yml (hunk 2 above)
Comment (kierenevans) [reply continuous-pipe.yml:44]: I think we do, as there is no wildcard DNS pointing at the cluster it's deployed to.

File: continuous-pipe.yml (hunk 3 above)
Comment (kierenevans) [reply continuous-pipe.yml:62]: Might be that it's set to true by default now (in :latest), but needed it for X-Forwarded-Proto detection for :stable.

**Pull request merged**